### ISSUE TITLE
Autocomplete on `$this->` does not work anymore

### ISSUE THREAD

**Issue by Taluu** [phpactor/phpactor#328]
After the last patches (can't determine which one messed it up), it seems that `$this->xyz` does not correctly complete and does not offer the accessible methods / properties.

```php
<?php
class Bar
{
    public function doSomething(): void
    {
    }
}

class Foo
{
    /** @var Bar */
    private $bar;

    public function __construct(Bar $bar)
    {
        // works
        $bar->....

        // does not work
        $this->bar = $bar;
        $this->bar->...
    }
}
```

I have the autocomplete working correctly for local variables, but not for properties.

**Comment by dantleech** [comment]
Are you aware that property completion is currently based on the saved version of the file? Currently it will try and locate a class (and file) called `Bar.php` somewhere, and it will fail.

Does this explain the behavior you are seeing?

**Comment by Taluu** [comment]
It is indeed made up, but this doesn't work even on separate saved files (one `Bar.php` and one `Foo.php`). Only the local variables work.

**Comment by Taluu** [comment]
Hum, it seems it works now that I committed the file that was having troubles.

It is strange though that the file *must* be committed on git to have its properties recognized (the local variables were working). That's all I did to make it work again.

**Comment by dantleech** [comment]
Yeah, that would explain it. This would basically be solved by adding a way to taking into account the current source file when _locating_ classes.

**Comment by dantleech** [comment]
See #332
